zephyr/include/rtos/string.h

```diff
@@ -34,24 +34,10 @@ static inline int rstrcmp(const char *s1, const char *s2)
 }
 
 /* C library does not have the "_s" versions used by Windows */
```

Comment:
to the commit message - sorry, which security concerns do you mean? The link is to a document about the Annex K itself, they do mention several potential security implications, did you mean any specific ones of them?

Reply:
First: this is off-topic. The only point relevant here is: "Zephyr does not have memcpy_s, most likely never will. SOF is on its own".

Now back to your question. https://www.open-std.org/jtc1/sc22/wg14/www/docs/n1969.htm has list of concerns about `memcpy_s()`, including some security concerns. I don't know which concerns in particular the Zephyr developers agree with, this is a question for the Zephyr project. I just heard some of them are listed in this document but I don't know which ones in particular.

Reply:
Without the _exact_ concerns pointed out I cannot in good faith offer a proper approve to this PR. I'm having mixed thoughts on whether memcpy_s is good or bad so I need more convincing to fully approve its removal and replacement with plain memcpy.

Reply:
Then this is too confusing for me to properly review. Why say that memcpy_s is a security concern, yet _don't_ remove it? The commit description and the actual action clash, so at least a different commit message is required.

Reply:
@paulstelian97 if you set aside (the bad parts of) the commit message for a moment, can you confirm you're OK with the code change?

Reply:
The change itself I'm fine if it builds.